sdks/java/core/src/main/java/org/apache/beam/sdk/io/AvroIO.java

```diff
@@ -142,6 +171,17 @@
   }
 
   /**
+   * Like {@link #readGenericRecords(Schema)}, but reads each filepattern in the input {@link
+   * PCollection}.
+   */
+  public static ReadAll<GenericRecord> readAllGenericRecords(Schema schema) {
```

Comment:
Though one issue here - AvroCoder does not support dynamic schemas, and for a source we need this to work....